OpenRA.Game/Network/Connection.cs

```diff
@@ -146,13 +148,14 @@ public virtual void SendSync(int frame, byte[] syncData)
 			var ms = new MemoryStream(4 + syncData.Length);
 			ms.WriteArray(BitConverter.GetBytes(frame));
 			ms.WriteArray(syncData);
-			Send(ms.GetBuffer());
+			Send(ms.ToArray());
```

Comment:
`ToArray` is a costly operation (https://github.com/microsoft/referencesource/blob/master/mscorlib/system/io/memorystream.cs line 550) and should be used only if needed. If `ms.Capacity == ms.Length` you should use `GetBuffer`